### DISCUSSION TITLE
[BUG] Error from OpenKeyChain : No encrypted data with known secret key found in stream!

### DISCUSSION THREAD

**Discussion by lucaspeixotot** [zeapo/Android-Password-Store#1310] [Q&A] [answered]
I'm using the Redmi Note 8T. I installed android-password-store, cloned the repository, installed the OpenKeyChain, linked it with password-store. My passwords are listed in password-store. After click on some of them, this error appears. 

If I go to OpenKeyChain > Apps > Password Store, there isn't keys allowed.

I created my keys in PC with "gpg --ful-generate-key --no-throw-keyids

How can I solve this?

**Comment by msfjarvis** [comment, marked as answer]
Hi, you need to import the private part of your GPG key into OpenKeychain to be able to decrypt passwords.

**Reply by lucaspeixotot** [reply]
How can I import it?

**Reply by msfjarvis** [reply]
Export your key using `gpg --armor --export-secret-keys key_id > private_key` (replace key_id with either the email used for the key or its actual ID), then copy `private_key` to your phone and use OpenKeychain's import from file option to select and add it. Once done, delete the key file from where you copied it to prevent unwanted access by other apps.

**Reply by marv42** [reply]
I imported my keypair and I checked the box corresponding to the key in the list of allowed keys of the Password Store app page.
But i am still getting the error as in the title of this discussion.
What else could be the problem?

**Reply by msfjarvis** [reply]
If the import was done correctly then the most likely option is that your store is encrypted with a different secret key.

**Reply by marv42** [reply]
Indeed! I was digging into password store documentation and found out which key i am using to encrypt the passwords. So, i realized that i have another key with the same email address.
Thanks!

**Comment by feelamee** [comment]
Hey. I faced with the same problem.
I have minimized the example and get rid of pass/Password store.
So, I have OpenPGP key, create with gpg.

My steps to reproduce:
- on my laptop
```bash
gpg --armor --export-secret-key fosterlounch@gmail.com > gpg.asc
echo "hello world" > hello-world
gpg -r fosterlounch@gmail.com -e hello-world
adb push hello-world.gpg gpg.asc /storage/self/primary
```
- on phone
  - import `gpg.asc` into OpenKeyChain. When I choose this key, it have green background. So, I think it imported correctly
  - choose Ecrypt/Decrypt in left-menu
  - select my file `hello-world.gpg`
  - take err "No ecrypted data with known secret key found in stream!"

May be I made a mistake somewhere?
I also tried vice versa - encrypt on phone and decrypt on laptop. And I noticed interesting detail. After encrypting on laptop with gpg I have file extension `.gpg`, but after encrypting with OpenKeyChain I have `.pgp`. Although, gpg on laptop successfully decrypt file, encrypted with OpenKeyChain.

Also I found [this](https://github.com/open-keychain/open-keychain/issues/2899) issue with pretty similar problem.

gpg version - 2.4.5
OpenKeyChain version - 6.0.4
Android Password Store - 1.13.5

I know that it not related to Password store, but maybe you have any solution for this?
By the way, likely, it will not be fixed, because OpenKeyChain no activecly developed now.
Do you have any plans to migrate on another OpenPGP client?

UPD:
I have seen [issue#1523](https://github.com/android-password-store/Android-Password-Store/issues/1523#issue-1034142844)
So, I will monitor it.

Thanks for you job!
You app is really useful
